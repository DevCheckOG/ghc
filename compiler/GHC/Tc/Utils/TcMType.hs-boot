module GHC.Tc.Utils.TcMType where

import GHC.Tc.Types
import GHC.Types.Name
import GHC.Core.TyCo.Rep

tcCheckUsage :: Name -> Mult -> TcM a -> TcM a
