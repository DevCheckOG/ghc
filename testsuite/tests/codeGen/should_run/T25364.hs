module Main where

import GHC.Utils.Encoding (zEncodeString,zDecodeString)
import Control.Monad

main :: IO ()
main = mapM_ test
  [ "ghc-prim_GHC.Types_$tc'(#,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,#)_closure"
  , "ghc-prim_GHC.Tuple_$tc'(,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,)2_closure"
  , "ghc-prim_GHC.Tuple_(,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,)_info"
  , "ghc-prim_GHC.Types_$tc'(# #)3_bytes"
  ]

test :: String -> IO ()
test s = do
  let e = zEncodeString s
  putStrLn e
  when (zDecodeString e /= s) $ do
    error $ "Invalid z-encoding roundtrip for: " ++ s
