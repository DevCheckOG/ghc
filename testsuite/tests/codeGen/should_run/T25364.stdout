ghczmprimzuGHCziTypeszuzdtczqZ32Hzuclosure
ghczmprimzuGHCziTuplezuzdtczqZ35T2zuclosure
ghczmprimzuGHCziTuplezuZ47Tzuinfo
ghczmprimzuGHCziTypeszuzdtczqZLzhz20UzhZR3zubytes
